tools/macosx/maple_upload

```diff
@@ -55,7 +55,7 @@ ${DFU_UTIL} -d ${usbID} -a ${altID} -D ${binfile} -R ${dfuse_addr} -R
 echo -n Waiting for ${dummy_port_fullpath} serial...
 
 COUNTER=0
-while [ ! -c ${dummy_port_fullpath} ] && ((COUNTER++ < 40)); do
+while ! dd if=${dummy_port_fullpath} of=/dev/null count=0 > /dev/null 2>&1 && ((COUNTER++ < 40)); do
```

Comment:
> 1. I mean, device existence should be enough, there's no need to try opening it for reading.

Because it does not have read permission though the device exist or you can't open the device for read though it have read permission. Please note that some device driver might return EACCESS on open operation even if the device entry on the file system have the permission. 

> 2. The same change is needed for linux (at least).

I can't test that modification because I don't have Linux PC. If you get device open error on your Linux, please try this patch.